### PR TITLE
refactor: format files with uncrustify

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

**Review by justinmk — APPROVED** [review]

File: src/nvim/ex_docmd.c [hunk 1]
```diff
@@ -2093,8 +2114,9 @@ int parse_command_modifiers(exarg_T *eap, char_u **errormsg, bool skip_only)
       cmdmod.split |= WSP_BOT;
       continue;
 
-    case 'c':   if (!checkforcmd(&eap->cmd, "confirm", 4))
+    case 'c':   if (!checkforcmd(&eap->cmd, "confirm", 4)) {
         break;
+    }
```
Comment (bfredl) [inline src/nvim/ex_docmd.c:2119]: this one a little weird. probably uncrustify failed to consider `if (expr) {` on the same line as `case 'c'` (and the fix is to move the `if` to the line below the `case` ).

Comment (bfredl) [reply src/nvim/ex_docmd.c:2119]: there's a few more cases like this, could be fixed manually with a suitable pattern for now :)

Comment (dundargoc) [reply src/nvim/ex_docmd.c:2119]: It's possible to add a newline after a case. Want me to try it out and add it as a separate commit so you can check if it looks good?

Comment (bfredl) [reply src/nvim/ex_docmd.c:2119]: yes.

Comment (dundargoc) [reply src/nvim/ex_docmd.c]: Added newline after case, but only if there's no brace after.

Comment (dundargoc) [reply src/nvim/ex_docmd.c]: While you're at it, could you take an extra look at the indentation on switch/case? I think there's currently no indentation between switch and case, is that correct?

Comment (justinmk) [reply src/nvim/ex_docmd.c]: > currently no indentation between switch and case, is that correct?

correct, though I don't think it matters much. Except that the current rule avoids some gratuitous indentation.

File: contrib/uncrustify.cfg [hunk 2]
```diff
@@ -1638,12 +1638,12 @@ nl_multi_line_define            = false    # true/false
...
 
 # Whether to add a newline after a 'case' statement.
-nl_after_case                   = false    # true/false
+nl_after_case                   = true    # true/false
 
 # Add or remove newline between a case ':' and '{'.
 #
 # Overrides nl_after_case.
-nl_case_colon_brace             = ignore   # ignore/add/remove/force/not_defined
+nl_case_colon_brace             = remove   # ignore/add/remove/force/not_defined
```
Comment (justinmk) [inline contrib/uncrustify.cfg:1646]: +1

**Review by justinmk — APPROVED** [review]

File: src/nvim/ops.c [hunk 3]
```diff
@@ -3409,23 +3474,27 @@ void do_put(int regname, yankreg_T *reg, int dir, long count, int flags)
...
               lendiff = (int)STRLEN(ptr);
-            if (*ptr == '#' && preprocs_left())
-              indent = 0;                   /* Leave # lines at start */
-            else if (*ptr == NUL)
-              indent = 0;                   /* Ignore empty lines */
-            else if (first_indent) {
+            }
+            if (*ptr == '#' && preprocs_left()) {
+              indent = 0;                   // Leave # lines at start
+            } else if (*ptr ==
```
Comment (bfredl) [inline src/nvim/ops.c:3482]: something went wrong here? :)

File: src/nvim/ex_docmd.c [hunk 4]
```diff
@@ -4987,28 +5089,23 @@ char_u *find_nextcmd(const char_u *p)
...
-static int
-check_more(
-    int message,                // when FALSE check only, no messages
-    bool forceit
-)
+/// - if there are more files to edit
+/// - and this is the last window
+/// - and forceit not used
+/// - and not repeated twice on a row
+///    return FAIL and give error message if 'message' TRUE, return OK otherwise
```
Comment (bfredl) [inline src/nvim/ex_docmd.c:5105]: I think you can add a `@return` here (I don't think there's any requirement to put `@return` below params, that's just a convention)

File: src/nvim/ex_docmd.c [hunk 5]
```diff
@@ -5726,25 +5839,18 @@ static size_t add_cmd_modifier(char_u *buf, char *mod_str, bool *multi_mods)
...
-    exarg_T *eap,               /* ex arguments */
-    char_u **split_buf,
-    size_t *split_len
-)
+/// Check for a <> code in a user command.
+/// "code" points to the '<'.  "len" the length of the <> (inclusive).
+/// "buf" is where the result is to be added.
+/// "split_buf" points to a buffer used for splitting, caller should free it.
+/// "split_len" is the length of what "split_buf" contains.
+/// Returns the length of the replacement, which has been added to "buf".
```
Comment (bfredl) [inline src/nvim/ex_docmd.c:5847]: same

File: src/nvim/ex_docmd.c [hunk 6]
```diff
@@ -8735,37 +8912,35 @@ ssize_t find_cmdline_var(const char_u *src, size_t *usedlen)
...
+///        '<cfile>' to path name under the cursor
+///        '<sfile>' to sourced file name
+///        '<slnum>' to sourced file line number
+///        '<afile>' to file name for autocommand
+///        '<abuf>'  to buffer number for autocommand
+///        '<amatch>' to matching name for autocommand
+///
+/// When an error is detected, "errormsg" is set to a non-NULL pointer (may be
+/// "" for error without a message) and NULL is returned.
+/// Returns an allocated string if a valid match was found.
```
Comment (bfredl) [inline src/nvim/ex_docmd.c:8932]: same

**Comment by bfredl** [comment]
LGTM after final comments :)

File: src/nvim/ex_docmd.c (hunk 4 above)
Comment (dundargoc) [reply src/nvim/ex_docmd.c:5105]: :+1:

File: src/nvim/ops.c (hunk 3 above)
Comment (dundargoc) [reply src/nvim/ops.c]: I've added this fix as a separate commit. Take a look and see if it looks alright.

File: src/nvim/ops.c [hunk 7]
```diff
@@ -2294,16 +2294,15 @@ void op_insert(oparg_T *oap, long count1)
         && !bd.is_MAX
         && !did_indent) {
       if (oap->op_type == OP_INSERT
-          && oap->start.col + oap->start.coladd
-          != curbuf->b_op_start_orig.col + curbuf->b_op_start_orig.coladd) {
+          && oap->start.col + oap->start.coladd !=
+          curbuf->b_op_start_orig.col + curbuf->b_op_start_orig.coladd) {
```
Comment (bfredl) [inline src/nvim/ops.c:2298]: this doesn't look right, if there _needs_ to be a linebreak, the comparisation operation should go on the new line, also ideally it should be intended within the `&&` .

File: src/nvim/ops.c (hunk 3 above)
Comment (dundargoc) [reply src/nvim/ops.c]: Uncrustify is better at _not_ splitting lines than joining them. I think the simplest fix would be to adjust this manually and uncrustify won't touch it afterwards. Is that alright solution?

Comment (bfredl) [reply src/nvim/ops.c:3482]: Sure let's be conservative and do it manually if uncrustify cannot be taught the right thing

Comment (dundargoc) [reply src/nvim/ops.c]: :+1:

**Comment by dundargoc** [comment]
@bfredl @justinmk Ok, done with all review comments.

**Comment by justinmk** [comment]
fyi for future PRs with big changes, it is very helpful to use "fixup" commits so that it is easy to see what changed

**Review by justinmk — APPROVED** [review]

**Comment by bfredl** [comment]
@justinmk to be fair he did, and squashed after I approved of the fixup changes.

**Comment by bfredl** [comment]
merged. there's a few funky lines left (around stuff like really long string constants) but let's not stall the **massive** reduction of `clint` noise that is in the balance.

**Comment by justinmk** [comment]
Might be better to squash-merge PRs like this, to avoid needing both the merge-commit and the actual commit in the .git-blame-ignore-revs

**Comment by dundargoc** [comment]
> fyi for future PRs with big changes, it is very helpful to use "fixup" commits so that it is easy to see what changed

Like @bfredl mentioned, I made a separate commit for each comment where I showed all the changes. After the changes were confirmed to be OK I squashed it to avoid excessive commits. Would you like me to skip squashing the fixup commits?

**Comment by Zeeshan376** [comment]
Plzz not sent email plzzzx

**Comment by clason** [comment]
> Plzz not sent email plzzzx
> […](#)

Don't watch this PR (or the Neovim repo in general) then.

**Comment by justinmk** [comment]
> Like @bfredl mentioned, I made a separate commit for each comment where I showed all the changes. 

@dundargoc yeah that was my mistake

> After the changes were confirmed to be OK I squashed it to avoid excessive commits. Would you like me to skip squashing the fixup commits?

Not necessary unless you have intentionally separated commits which should be merged instead of squash-merged. Otherwise, whoever merges the PR should use the squash-merge option.

**Pull request merged**